spec/controllers/calls_controller_spec.rb

```diff
@@ -2,11 +2,49 @@
 
 describe CallsController,  type: :controller do
 
-  def setup_active_call_double(target_legislators:[])
+  def setup_active_call_double(target_legislators:[], new_record:false)
     @active_call = double('active_call')
+    person = double('person')
+    allow(@active_call).to receive(:person).and_return(!new_record && person || nil)
+    allow(@active_call).to receive(:new_record?).and_return(new_record)
     allow(@active_call).to receive(:target_legislators).and_return(target_legislators)
     allow(@active_call).to receive(:exceeded_max_connections?).and_return(false)
-    allow(Ivr::Call).to receive_message_chain(:includes, :where, :first_or_create).and_return(@active_call)
+    allow(Ivr::Call).to receive_message_chain(:includes, :first_or_initialize).and_return(@active_call)
+  end
+
+  def setup_last_connection
+    @last_connection = double('last_connection')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.